pkg/util/clusteragent/clcrunner.go

```diff
@@ -0,0 +1,161 @@
+	clcRunnerStatsPath   = "stats"
+)
+
+var globalCLCRunnerClient *CLCRunnerClient
+
+// CLCRunnerClientInterface is required to query the API of Datadog Cluster Level Check Runner
+type CLCRunnerClientInterface interface {
+	GetVersion(IP string) (version.Version, error)
+	GetRunnerStats(IP string) (types.CLCRunnersStats, error)
+}
+
+// CLCRunnerClient is required to query the API of Datadog Cluster Level Check Runner
+type CLCRunnerClient struct {
+	clcRunnerAPIRequestHeaders http.Header
+	clcRunnerAPIClient         *http.Client
+	clcRunnerPort              int
+}
+
+// resetGlobalCLCRunnerClient is a helper to remove the current CLCRunnerClient global
+// It is ONLY to be used for tests
+func resetGlobalCLCRunnerClient() {
+	globalCLCRunnerClient = &CLCRunnerClient{}
+	globalCLCRunnerClient.init()
+}
+
+// GetCLCRunnerClient returns or init the CLCRunnerClient
+func GetCLCRunnerClient() (CLCRunnerClientInterface, error) {
+	if globalCLCRunnerClient == nil {
+		globalCLCRunnerClient = &CLCRunnerClient{}
+		if err := globalCLCRunnerClient.init(); err != nil {
+			log.Debugf("CLC Runner client init error: %v", err)
+			return nil, err
+		}
+	}
+	return globalCLCRunnerClient, nil
+}
+
+func (c *CLCRunnerClient) init() error {
+	authToken, err := security.GetClusterAgentAuthToken()
+	if err != nil {
+		return err
+	}
+
+	// Set headers
+	c.clcRunnerAPIRequestHeaders = http.Header{}
+	c.clcRunnerAPIRequestHeaders.Set(authorizationHeaderKey, fmt.Sprintf("Bearer %s", authToken))
+
+	// Set http client
+	// TODO remove insecure
+	c.clcRunnerAPIClient = util.GetClient(false)
+	c.clcRunnerAPIClient.Timeout = 2 * time.Second
+
+	// Set http port used by the CLC Runners
+	c.clcRunnerPort = config.Datadog.GetInt("cmd_port")
+
+	return nil
+}
+
+// GetVersion fetches the version of the CLC Runner
+func (c *CLCRunnerClient) GetVersion(IP string) (version.Version, error) {
+	var version version.Version
+	var err error
+
+	rawURL := fmt.Sprintf("https://%s:%d/%s/%s", IP, c.clcRunnerPort, clcRunnerPath, clcRunnerVersionPath)
+
+	req, err := http.NewRequest("GET", rawURL, nil)
+	if err != nil {
+		return version, err
+	}
+	req.Header = c.clcRunnerAPIRequestHeaders
+
+	resp, err := c.clcRunnerAPIClient.Do(req)
+	if err != nil {
+		return version, err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		return version, fmt.Errorf("unexpected status code from CLC runner: %d", resp.StatusCode)
+	}
+
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return version, err
+	}
+
+	err = json.Unmarshal(body, &version)
+
+	return version, err
+}
+
+// GetRunnerStats fetches the runner stats exposed by the Cluster Level Check Runner
+func (c *CLCRunnerClient) GetRunnerStats(IP string) (types.CLCRunnersStats, error) {
+	var stats types.CLCRunnersStats
+	var err error
+
+	rawURL := fmt.Sprintf("https://%s:%d/%s/%s", IP, c.clcRunnerPort, clcRunnerPath, clcRunnerStatsPath)
+
+	req, err := http.NewRequest("GET", rawURL, nil)
```

Comment:
I think it can be good to set a timeout for this request.

Reply:
we set it in L79